src/spauna.ts

```diff
@@ -1,24 +1,79 @@
 // See https://stackoverflow.com/a/74112582
 /// <reference lib="dom" />
+import { readFile } from "node:fs/promises";
+import path from "node:path";
+import { pathToFileURL } from "node:url";
 
 export async function runCommand(
   commandName: string,
   configLocation: string,
   args: object,
 ) {
   // Step 1: Fetch Config
+  const config = await fetchConfig(
+    configLocation,
+    pathToFileURL(process.cwd() + "/"),
+  );
+
+  console.log(config);
   // Step 2: Find Command
   // Step 3: Execute Command
 }
 
-async function fetchConfig(configLocation: string) {
+interface SpaunaConfig {
+  extends?: string;
+}
+
+async function fetchConfig(
+  configLocation: string,
+  parentUrl: URL,
+): Promise<SpaunaConfig> {
+  let config: SpaunaConfig;
+  let extendParent: URL = parentUrl;
+
   if (
     configLocation.startsWith("http://") ||
     configLocation.startsWith("https://")
   ) {
     const configResult: Response = await fetch(configLocation);
     if (configResult.ok) {
-      const json = await configResult.json();
+      config = await configResult.json();
     }
+  } else if (
+    configLocation.startsWith("./") ||
+    configLocation.startsWith("../") ||
+    configLocation.startsWith("file://")
+  ) {
+    let fileUrl: URL;
+
+    if (configLocation.startsWith("file://")) {
+      fileUrl = new URL(configLocation);
+    } else {
+      fileUrl = new URL(configLocation, parentUrl);
+    }
```

Comment:
This could be simplified to:

```suggestion
    let fileUrl: URL = new URL(configLocation, parentUrl);
```

Since `parentUrl` is ignored if a full URL is provided.

Reply:
Good catch, I've updated this